src/arrayMethodSort.js

```diff
@@ -4,8 +4,41 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
+  [].__proto__.sort2 = function (compareFunction) {
     // write code here
+    const newCompareFunc = (a, b) => {
+      let firstItem = a;
+      let secondItem = b;
+      
+      if (typeof a === 'number' && typeof b === 'number') {
+        firstItem = String(a);
+        secondItem = String(b)
+      }
+
+      if (firstItem < secondItem) {
+        return -1;
+      }
+
+      if (firstItem > secondItem) {
+        return 1;
+      }
+
+      return 0;
+    };
```

Comment:
The custom comparison function `newCompareFunc` is converting numbers to strings before comparing them, which is unnecessary. The default sort behavior for numbers should be to compare their values directly. Consider removing the conversion to strings and simply comparing the numbers.